src/Compilers/VisualBasic/Portable/Symbols/ArrayTypeSymbol.vb

```diff
@@ -314,34 +314,36 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         Friend MustOverride Overrides Function InternalSubstituteTypeParameters(substitution As TypeSubstitution) As TypeWithModifiers
 
         Public Overrides Function Equals(obj As Object) As Boolean
+            Return IsSameType(obj, TypeCompareKind.ConsiderEverything)
+        End Function
+
+        Friend Function IsSameType(obj As Object, compareKind As TypeCompareKind) As Boolean
+            Debug.Assert((compareKind And Not (TypeCompareKind.IgnoreCustomModifiersAndArraySizesAndLowerBounds Or TypeCompareKind.IgnoreTupleNames)) = 0)
```

Comment:
I added this option to the enum, but I prefer to leave an explicit list here, as it captures that this method is designed to work with those known options.

---
In reply to: [81685354](https://github.com/dotnet/roslyn/pull/14229#discussion_r81685354) [](ancestors = 81685354)